### PR TITLE
Fix #12143: MenuButton remove duplicate inherited code

### LINKED ISSUE #12143: MenuButton: not disappearing after clicking in PF 14.0.1
### Describe the bug

Between PF 14.0.0 and 14.0.1 the behavior of menuItem inside menuButton changed. In 14.0.0 clicking a menuItem makes the menu disappear. In PF 14.0.1 the menu stays open and you have to click somewhere outside the menu (so it looses focus) to make it disappear.

Example xhtml pasted below. Please run the file in any browser (tested on firefox, opera and chromium on Kubuntu 24.04 and Windows 10) in PF 14.0.0 and 14.0.1 to see the difference in menu behavior.

You can also run attached Spring Boot project (pf-demo.zip) and go to localhost:8080/menu.xhtml

### Reproducer

```html
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html lang="en-us"
      xmlns:h="jakarta.faces.html"
      xmlns:p="http://primefaces.org/ui">

<h:head>
    <title>.:: Primefaces with JoinFaces Example ::.</title>
    <meta content='text/html; charset=UTF-8' http-equiv="Content-Type"/>
</h:head>

<h:body>
    <h:form>
        <p:menuButton id="helpButton" value="Help button" title="Help">
            <p:menuitem value="Help1" title="Help F1" target="_blank">

            </p:menuitem>
            <p:menuitem value="Help2" title="Help2" target="_blank">

            </p:menuitem>
        </p:menuButton>
    </h:form>
</h:body>

</html>
```
[pf-demo.zip](https://github.com/user-attachments/files/15902831/pf-demo.zip)

### Expected behavior

Clicking a menuItem should make the menu disappear.

### PrimeFaces edition

Community

### PrimeFaces version

14.0.1

### Theme

_No response_

### JSF implementation

Mojarra

### JSF version

4.0 (not sure what PF and JoinFaces use in Spring Boot project)

### Java version

17.0.11

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #12143: MenuButton remove duplicate inherited code

### CODE REVIEW

File: primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js [hunk 1]
```diff
@@ -136,19 +136,20 @@ PrimeFaces.widget.Menu = PrimeFaces.widget.BaseWidget.extend({
      */
     bindPanelEvents: function() {
         var $this = this;
+        var $menu = this.menu || this.jq;
```
Comment (tandraschko) [inline primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js]: i dont like this TBH, its a bit dirty
we should add a method like getMenuElement and overwrite it

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js]: ok got it!

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js]: Refactored to a `getMenu()` method that is overridden in MenuButton.

File: primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js [hunk 2]
```diff
@@ -45,6 +44,15 @@ PrimeFaces.widget.Menu = PrimeFaces.widget.BaseWidget.extend({
         }
     },
 
+    /**
+     * Gets the Menu jQuery element.  Override in subclasses to define the menu panel.
+     * @returns {JQuery} The jQuery object for the menu.
+     * @protected
+     */
+    getMenu: function() {
```
Comment (tandraschko) [inline primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js]: would really call it getMenuElement

the menu is the whole component itself already

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js]: Renamed to `getMenuElement()`

**Comment by vonnai** [comment]
Hi, your optimiziation with getMenuElement method is not working well. You must always call ```this.menu = this.jq.children('.ui-menu');``` in the init method, otherwise the menu reference will lose the parent after refresh. Don't forget that there is called cleanupDynamicOverlay, that removes the overlay from the body during refresh the widget.
Thank you.

**Comment by melloware** [comment]
@vonnai already been reported and fixed for 14.0.3: https://github.com/primefaces/primefaces/pull/12289

**Pull request merged**